components/header.tsx

```diff
@@ -1,43 +1,171 @@
 import { useLocale } from '@/lib/hooks'
-import { CommonType } from '@/lib/interface'
+import { useState, useEffect } from 'react'
 import Link from 'next/link'
+
+import { CommonType } from '@/lib/interface'
+import { LanguagesType } from '@/lib/interface'
+
 import { ChangeLangButton } from './data-components/change-language-button'
 import OneZeroSkipToMainContent from './onezero-skip-to-main-content'
 
 export default function Header({ data }: CommonType) {
   const { dir } = useLocale()
+  const [languages, setLanguages] = useState<any>([])
+  const [isMenuOpen, setIsMenuOpen] = useState<any>(false)
+
+  useEffect(() => {
+    const langObj: LanguagesType = data.languageNames[0]
+    const langArr = Object.keys(langObj).map((key) => key)
+    const ar = langArr[2]
+    langArr.splice(2, 1)
+    langArr.unshift(ar)
+    setLanguages(langArr)
+  }, [data])
+
+  const toggleMenu = () => {
+    // document.body.classList.toggle('menu-open')
+    setIsMenuOpen(!isMenuOpen)
+  }
+
+  if (!data || !languages) return <div></div>
   return (
     <>
       <OneZeroSkipToMainContent
         text={'skipToMainContent'}
         dir={dir}
         className={'bg-light text-primary'}
       />
-      <header className="h-header z-10 pt-4 px-4">
-        <div className="grid grid-cols-auto-1fr gap-x-6 mx-auto max-w-screen-lg p-3 bg-light rounded-lg">
-          <Logo />
-          <ul className="flex flex-row gap-x-2">
-            <li>
-              <ChangeLangButton className="" lang="en">
-                {data.languageNames[0].en}
-              </ChangeLangButton>
-            </li>
-            <li>
-              <ChangeLangButton className="" lang="he">
-                {data.languageNames[0].he}
-              </ChangeLangButton>
-            </li>
-          </ul>
+      <header className="absolute w-full z-10 sm:pt-3 sm:px-3 pt-1 px-1">
+        <div className="sm:grid sm:grid-cols-auto-1fr sm:gap-x-6 sm:mx-auto sm:max-w-screen-lg sm:py-1 sm:px-3 bg-light rounded-lg h-22 sm:items-center sm:relative">
+          <div className="flex justify-between p-1 sm:p-0">
```

Comment:
hamburger menu looks like a good candidate for a separate component. why clutter the header with so much html css and functionality?

Reply:
done